working/augmentation-libraries/feature-specification.md

```diff
@@ -321,82 +322,150 @@ doesn't, it must repeat the type from the augmented definition__.
+being augmented.
+
+*   **Augmenting variables**: Within an augmenting variable declaration's
+    initializer expression, `augmented` can only be used in an initializer
+    expression, and it refers to the augmented variable's initializing
+    expression. The reserved words parses as a `<primary>` expression.
+    Evaluating `augmented` evaluates the augmented declaration's initializer
+    expression, and evaluates to the same result.
+    The static type of `augmented` is the static type of the initializer
+    expression of the augmented declaration.
+
+*   **Augmenting getters**: Within an augmenting getter's body, the `augmented`
+    expression invokes the augmented getter and evaluates to its return value.
+    When augmenting a variable with a getter, this will invoke the implicitly
+    induced getter of the augmented variable declaration.
+    _This applies even to `late` variable declarations, where the implicit
+    getter does more than just access the underlying storage._
+    The reserved words parses as a `<primary>` expression.
+    Evaluating `augmented` invokes the augmented declaration's getter
+    function, and evaluates to the returned value.
+    The static type of `augmented` is the declared return type of the augmented
+    declaration.
+
+*   **Augmenting setters**: Within an augmenting setter's body, `augmented`
+    must be followed by an `= expression`.
+    _The `augmented` term is not an expression that evaluates to a value.
+    Like `super` it is a special syntactic form that can only be used
+    in specific ways. Except for the prior two cases, where that specific way
+    was to be read and evaluate to a value, `augmented` cannot be used
+    as an expression with a value._
+    The clause <code>augmented = *e*</code> parses as an `<expression>`.
+    Performing the assignment of the assignment expression, after evaluating
+    the right-hand-side to a value, invokes the augmented setter function.
+    When augmenting an assignable variable declaration, that setter function
+    is the implicitly induced setter of the variable declaration.
+    _This also applies to `late` and `late final` variable declarations._
+    Type inference of `augmented = e1` is performed as a setter invocation
+    with the setter signature of the augmented declaration.
 
 *   **Augmenting functions**: When augmenting a function, `augmented`
-    refers to the augmented function. Tear offs are not allowed, so this
-    function must immediately be invoked.
-
-*   **Augmenting operators**: When augmenting an operator, `augmented`
-    refers to the augmented operator method, which must be immediately
-    invoked using function call syntax. For example, when augmenting
+    in the function body refers to the augmented function, and it must
+    occur only as the function of a function invocation, which means
+    it is immediately followed by one, or no, type argument list
+    and then an argument list.
```

Comment:
This is a little difficult to read. Maybe:

"which means it is immediately followed by a type argument list (which may be omitted) then an argument list (which must be present but may be empty)."

Reply:
I went through a lot of phrasings here, including "zero or one type parameter lists", but the plural annoyed me.
Maybe just:
> it is immediately followed by an argument list, or a type parameter list and then an argument list.